rust/arrow/src/json/reader.rs

```diff
@@ -250,19 +326,17 @@ pub fn infer_json_schema<R: Read>(
     reader: &mut BufReader<R>,
     max_read_records: Option<usize>,
 ) -> Result<SchemaRef> {
-    let mut values: HashMap<String, HashSet<DataType>> = HashMap::new();
-
-    let mut line = String::new();
-    for _ in 0..max_read_records.unwrap_or(std::usize::MAX) {
-        reader.read_line(&mut line)?;
-        if line.is_empty() {
-            break;
-        }
-        let record: Value = serde_json::from_str(&line.trim()).expect("Not valid JSON");
```

Comment:
is this the line that causes a hard crash on invalid json?

Reply:
that's right.